includes/html/api_functions.inc.php

```diff
@@ -2838,6 +2838,21 @@ function edit_location(Illuminate\Http\Request $request)
     return api_error(500, 'Failed to update location');
 }
 
+function get_location(Illuminate\Http\Request $request)
+{
+    $location = $request->route('location_id_or_name');
+    if (empty($location)) {
+        return api_error(400, 'No location has been provided to get');
+    }
+    $location_id = ctype_digit($location) ? $location : get_location_id_by_name($location);
+    $data = dbFetchRows('SELECT `locations`.* FROM `locations` WHERE `locations`.`id` = ?', [$location_id]);
```

Comment:
You should not use dbFetchRows, it is deprecated.
```php
$data = \App\Models\Location::find($location_id);
```

You could also prevent the get_location_id_by_name() name above with a more complex query.  (that function calls the database)